PLUB/Sources/Views/Home/Component/HomeAlert.swift

```diff
@@ -75,53 +92,133 @@ final class HomeAlert {
+  
+  public func showAlert(type: HomeAlertType = .applyCompleted) {
     guard let keyWindow = UIApplication.shared.connectedScenes
       .filter({$0.activationState == .foregroundActive})
       .compactMap({$0 as? UIWindowScene})
       .first?.windows
       .filter({$0.isKeyWindow}).first else { return }
     
+    alertView.removeFromSuperview()
     [backgroundView, alertView].forEach { keyWindow.addSubview($0) }
-    [backButton, stackView].forEach { alertView.addSubview($0) }
-    
     backgroundView.alpha = Constants.backgroundAlphaTo
     backgroundView.snp.makeConstraints {
       $0.edges.equalToSuperview()
     }
     
-    alertView.snp.makeConstraints {
-      $0.center.equalToSuperview()
-      $0.width.equalTo(296)
-      $0.height.equalTo(448)
-    }
-    
-    backButton.snp.makeConstraints {
-      $0.width.height.equalTo(32)
-      $0.top.trailing.equalToSuperview().inset(10)
-    }
+    switch type {
+    case .applyCompleted:
+      [backButton, stackView].forEach { alertView.addSubview($0) }
+      
+      alertView.snp.makeConstraints {
+        $0.center.equalToSuperview()
+        $0.width.equalTo(296)
+        $0.height.equalTo(448)
+      }
+      
+      backButton.snp.makeConstraints {
+        $0.width.height.equalTo(32)
+        $0.top.trailing.equalToSuperview().inset(10)
+      }
+      
+      stackView.snp.makeConstraints {
+        $0.top.equalTo(backButton.snp.bottom).offset(0.5)
+        $0.leading.trailing.bottom.equalToSuperview()
+      }
+      
+      stackView.setCustomSpacing(8, after: mainLabel)
+      stackView.setCustomSpacing(32, after: subLabel)
+    case .cancelApply:
+      [mainLabel, buttonStackView].forEach { alertView.addSubview($0) }
+      alertView.snp.makeConstraints {
+        $0.center.equalToSuperview()
+        $0.width.equalTo(296)
+        $0.height.equalTo(210)
+      }
     
-    stackView.snp.makeConstraints {
-      $0.top.equalTo(backButton.snp.bottom).offset(0.5)
-      $0.leading.trailing.bottom.equalToSuperview()
+      mainLabel.text = "이 모임에 참여하고 싶지\n 않으신가요?"
+      mainLabel.snp.makeConstraints {
+        $0.top.equalToSuperview().inset(59)
+        $0.directionalHorizontalEdges.equalToSuperview().inset(48)
+      }
+      
+      buttonStackView.snp.makeConstraints {
+        $0.directionalHorizontalEdges.bottom.equalToSuperview().inset(16)
+        $0.height.equalTo(46)
+      }
+      
     }
     
-    stackView.setCustomSpacing(8, after: mainLabel)
-    stackView.setCustomSpacing(32, after: subLabel)
   }
   
-  @objc private func dismissAlert() {
+  private func dismissAlert() {
     UIView.animate(withDuration: 0.25) {
       self.backgroundView.alpha = 0
     } completion: { done in
       if done {
         self.alertView.removeFromSuperview()
         self.backgroundView.removeFromSuperview()
+        self.delegate?.didTappedCancelButton()
       }
     }
   }
 }
+
+extension HomeAlert {
+  struct Constants {
+    static let backgroundAlphaTo = 0.6
+  }
```

Comment:
단순 의견: enum으로 바꾸는 것도 좋은 전략 중 하나일 것 같아요.
개발자 실수로 인스턴스 되는 것을 막아주니까요.
```suggestion
  enum Constants {
    static let backgroundAlphaTo = 0.6
  }
```

Reply:
오 바로 야곰님 의견 수용하셨군요 좋은거같아요 !!